lib/active_shipping/carriers/canada_post_pws.rb

```diff
@@ -442,102 +449,107 @@ def shipment_customs_node(destination, line_items, options)
-      doc = REXML::Document.new(response)
-      root = doc.elements['non-contract-shipment-receipt']
-      cc_details_node = root.elements['cc-receipt-details']
-      service_standard_node = root.elements['service-standard']
+      doc = Nokogiri.XML(response)
+      doc.remove_namespaces!
+      root = doc.at('non-contract-shipment-receipt')
+      cc_details_node = root.at('cc-receipt-details')
+      service_standard_node = root.at('service-standard')
       receipt = {
-        :final_shipping_point => root.get_text("final-shipping-point").to_s,
-        :shipping_point_name => root.get_text("shipping-point-name").to_s,
-        :service_code => root.get_text("service-code").to_s,
-        :rated_weight => root.get_text("rated-weight").to_s.to_f,
-        :base_amount => root.get_text("base-amount").to_s.to_f,
-        :pre_tax_amount => root.get_text("pre-tax-amount").to_s.to_f,
-        :gst_amount => root.get_text("gst-amount").to_s.to_f,
-        :pst_amount => root.get_text("pst-amount").to_s.to_f,
-        :hst_amount => root.get_text("hst-amount").to_s.to_f,
-        :charge_amount => cc_details_node.get_text("charge-amount").to_s.to_f,
-        :currency => cc_details_node.get_text("currency").to_s,
-        :expected_transit_days => service_standard_node.get_text("expected-transit-time").to_s.to_i,
-        :expected_delivery_date => service_standard_node.get_text("expected-delivery-date").to_s
+        :final_shipping_point => root.at("final-shipping-point").text,
+        :shipping_point_name => root.at("shipping-point-name").text,
+        :service_code => root.at("service-code").text,
+        :rated_weight => root.at("rated-weight").text.to_f,
+        :base_amount => root.at("base-amount").text.to_f,
+        :pre_tax_amount => root.at("pre-tax-amount").text.to_f,
+        :gst_amount => root.at("gst-amount").text.to_f,
+        :pst_amount => root.at("pst-amount").text.to_f,
+        :hst_amount => root.at("hst-amount").text.to_f,
+        :charge_amount => cc_details_node.at("charge-amount").text.to_f,
+        :currency => cc_details_node.at("currency").text,
+        :expected_transit_days => service_standard_node.at("expected-transit-time").text.to_i,
+        :expected_delivery_date => service_standard_node.at("expected-delivery-date").text
       }
-      option_nodes = root.elements['priced-options'].elements.collect('priced-option') { |node| node } unless root.elements['priced-options'].blank?
+      option_nodes = root.xpath('priced-options/priced-option')
 
-      receipt[:priced_options] = if option_nodes
-          option_nodes.inject({}) do |result, node|
-          result[node.get_text("option-code").to_s] = node.get_text("option-price").to_s.to_f
+      receipt[:priced_options] = if option_nodes.length > 0
+        option_nodes.inject({}) do |result, node|
+          result[node.at("option-code").text] = node.at("option-price").text.to_f
           result
           end
       else
-        []
+        {}
```

Comment:
This was a small bug in the original implementation. The `if` clause is returning a hash, but the `else` clause is returning an array. Given that it's empty, that probably never caused any problems, but it irked me.